src/cunumeric/index/select.cu

```diff
@@ -0,0 +1,122 @@
+/* Copyright 2023 NVIDIA Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "cunumeric/index/select.h"
+#include "cunumeric/index/select_template.inl"
+#include "cunumeric/cuda_help.h"
+
+namespace cunumeric {
+
+template <typename VAL>
+__global__ static void __launch_bounds__(THREADS_PER_BLOCK, MIN_CTAS_PER_SM)
+  select_kernel_dense(VAL* outptr,
+                      uint32_t narrays,
+                      legate::Buffer<const bool*, 1> condlist,
+                      legate::Buffer<const VAL*, 1> choicelist,
+                      VAL default_val,
+                      int volume)
+{
+  const size_t idx = global_tid_1d();
+  if (idx >= volume) return;
+  outptr[idx] = default_val;
+  for (int32_t c = (narrays - 1); c >= 0; c--) {
+    if (condlist[c][idx]) { outptr[idx] = choicelist[c][idx]; }
+  }
+}
+
+template <typename VAL, int DIM>
+__global__ static void __launch_bounds__(THREADS_PER_BLOCK, MIN_CTAS_PER_SM)
+  select_kernel(const AccessorWO<VAL, DIM> out,
+                uint32_t narrays,
+                const legate::Buffer<AccessorRO<bool, DIM>> condlist,
+                const legate::Buffer<AccessorRO<VAL, DIM>> choicelist,
+                VAL default_val,
+                const Rect<DIM> rect,
+                const Pitches<DIM - 1> pitches,
+                int out_size,
+                int volume)
+{
+  const size_t tid = global_tid_1d();
+  if (tid >= out_size) return;
```

Comment:
Can you do this like the dense case, with one thread per element (rather than how it's done now, with each thread managing multiple elements)?